internal/experiments/chaos_controller/chaosmesh.go

```diff
@@ -0,0 +1,232 @@
+package chaos_controller
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+	"xk6-khorne/internal/k8s/pods"
+	"xk6-khorne/pkg/k8s/client"
+
+	coreV1 "k8s.io/api/core/v1"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+type ChaosController struct {
+	cluster_config_path string
+	pod                 *pods.Pods
+	dynamicClient       *dynamic.DynamicClient
+	namespace           string
+	curExp              *unstructured.Unstructured
+	gvr                 *schema.GroupVersionResource
+}
+
+// New creates a new podkiller
+func New(namespace string) *ChaosController {
+	c, err := client.New()
+	p := pods.New(c)
+	if err != nil {
+		return nil
+	}
+
+	config_path := getConfigPath()
+
+	config, err := clientcmd.BuildConfigFromFlags("", config_path)
+	if err != nil {
+		log.Fatalf("Failed to build kubeconfig: %v", err)
+	}
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("Failed to create Kubernetes client: %v", err)
+	}
+
+	return &ChaosController{config_path, p, dynamicClient, namespace, nil, nil}
+}
+
+// configPath    - path to k8s cluster config
+// killNodes 	- to kill or not to kill affected nodes after each run
+// recoveryTime - how much time to wait for cluster to recover after experiment
+func (cc *ChaosController) RunChaosExperiment(configPath string, restartNodes bool, recoveryTime string) error {
+	// Work with experiment yaml file
+	experiment_file, err := os.ReadFile(configPath)
+	if err != nil {
+		log.Fatalf("Failed to read manifest file: %v", err)
+	}
+	reader := bytes.NewReader(experiment_file)
+	decoder := yaml.NewYAMLOrJSONDecoder(reader, 1000)
+	parsedYaml := &unstructured.Unstructured{}
```

Comment:
Maybe you can use this struct https://github.com/chaos-mesh/chaos-mesh/blob/release-2.7/api/v1alpha1/networkchaos_types.go#L31 for parsing?

Reply:
I think so, however existing structure seems more general. There will be another chaos types (podChaos for example).